### DISCUSSION TITLE
Trying to add a telescope extension but not sure how to call it

### DISCUSSION THREAD

**Discussion by neonvoidx** [LazyVim/LazyVim#1918] [Q&A] [answered]
I am trying to add the key here:
`    { "<leader>/", require("lazyvim.util").telescope("live_grep_args").live_grep_args(), desc = "Grep (root dir)" },`
for this plugin https://github.com/nvim-telescope/telescope-live-grep-args.nvim. I want to replace default live_grep binding with this because it allows to grep with things like file name filters etc. The docs for plugin say to use `require("telescope").extensions.live_grep_args.live_grep_args()` but that doesn't work either

Full telescope override here:
```
return {
  "nvim-telescope/telescope.nvim",
  keys = {
    { "<leader><space>", require("lazyvim.util").telescope("files", { cwd = false }), desc = "Find Files (cwd)" },
    { "<leader>fF", require("lazyvim.util").telescope "files", desc = "Find Files (root dir)" },
    { "<leader>ff", require("lazyvim.util").telescope("files", { cwd = false }), desc = "Find Files (cwd)" },
    { "<leader>/", require("lazyvim.util").telescope("live_grep_args").live_grep_args(), desc = "Grep (root dir)" },
    -- { "<leader>sg", require("lazyvim.util").telescope "live_grep", desc = "Grep (root dir)" },
    -- { "<leader>sG", require("lazyvim.util").telescope("live_grep", { cwd = false }), desc = "Grep (cwd)" },
  },
  opts = {
    defaults = {
      layout_strategy = "vertical",
      layout_config = { prompt_position = "top", width = 0.9, height = 0.9 },
      sorting_strategy = "ascending",
      winblend = 0,
    },
  },
  dependencies = {
    {
      "nvim-telescope/telescope-live-grep-args.nvim",
    },
    {
      "nvim-telescope/telescope-fzf-native.nvim",
      build = "make",
      config = function()
        local tele = require "telescope"
        tele.load_extension "fzf"
        tele.load_extension "bookmarks"
        tele.load_extension "workspaces"
        tele.load_extension "live_grep_args"
      end,
    },
  },
}

```

**Comment by dpetka2001** [comment]
The `config` function you're using is residing in the `telescope-fzf-native` spec. Instead the `config` function should reside in  `nvim-telescope/telescope.nvim`'s spec itself and all the `load_extension` part should happen after `require("telescope").setup()` itself. You can take a look at my dotfiles [Telescope](https://github.com/dpetka2001/dotfiles/blob/main/dot_config/nvim/lua/plugins/telescope.lua) for how it should be structured.

**Reply by dpetka2001** [reply, marked as answer]
A simple config based on what you provided in your initial post would be
```lua
return {
  {
    "nvim-telescope/telescope.nvim",
    dependencies = {
      {
        "nvim-telescope/telescope-live-grep-args.nvim",
      },
    },
    opts = {
      defaults = {
        layout_strategy = "vertical",
        layout_config = { prompt_position = "top", width = 0.9, height = 0.9 },
        sorting_strategy = "ascending",
        winblend = 0,
      },
    },
    keys = {
      {
        "<leader>/",
        "<cmd>lua require('telescope').extensions.live_grep_args.live_grep_args()<CR>",
        desc = "Grep (root dir)",
      },
    },
    config = function(_, opts)
      local tele = require("telescope")
      tele.setup(opts)
      tele.load_extension("live_grep_args")
    end,
  },
}
```
I didn't include `fzf-native` or anything else, cuz I tried to keep it to a minimum for `telescope-live-args` only and you can add whatever else you would like.

**Reply by neonvoidx** [reply]
Wondering if you know the answer to this part, i want to add the mappings for live_grep_args, in docs it says to:
```
local telescope = require("telescope")
local lga_actions = require("telescope-live-grep-args.actions")

telescope.setup {
  extensions = {
    live_grep_args = {
      auto_quoting = true, -- enable/disable auto-quoting
      -- define mappings, e.g.
      mappings = { -- extend mappings
        i = {
          ["<C-k>"] = lga_actions.quote_prompt(),
          ["<C-i>"] = lga_actions.quote_prompt({ postfix = " --iglob " }),
        },
      },
      -- ... also accepts theme settings, for example:
      -- theme = "dropdown", -- use dropdown theme
      -- theme = { }, -- use own theme spec
      -- layout_config = { mirror=true }, -- mirror preview pane
    }
  }
}
```
 I assume I'd put that into the `opts` field for telescope, but I know you should require things in opts correct? so do i have to call it in the config portion as well? i dont want to override the default LazyVim telescope setup

**Reply by neonvoidx** [reply]
seems to work with the telescope.setup put into the config function portion, doesnt seem to override defaults

**Reply by dpetka2001** [reply]
You can also just put it in the telescope `opts` section.
